### PR TITLE
building a constant_sequence is O(log N) instead of O(N) in template …

### PR DESCRIPTION
**Pull request by ericniebler**
…instantiations

### CODE REVIEW

File: fatal/type/sequence.h [hunk 1]
```diff
@@ -836,7 +837,31 @@ using size_sequence = constant_sequence<std::size_t, Values...>;
 namespace detail {
 namespace constant_sequence_impl {
 
-template <bool, bool, typename T, T, T> struct build;
+enum class Flag { eDone, eDouble, eRecurse };
```
Comment (juchem) [inline fatal/type/sequence.h]: Fatal uses a different naming convention, similar to the standard library's

File: fatal/type/sequence.h [hunk 2]
```diff
@@ -836,7 +837,31 @@ using size_sequence = constant_sequence<std::size_t, Values...>;
...
+
+constexpr Flag flag(std::size_t cur, std::size_t end) {
+  return cur >= end ? Flag::eDone
```
Comment (juchem) [inline fatal/type/sequence.h]: please, adopt the format conventions as in the rest of Fatal:

```
return cur >= end
  ? build_strategy::done
  : cur * 2 <= end
    ? build_strategy::split
    : build_strategy::recurse;
```

File: fatal/type/sequence.h [hunk 3]
```diff
@@ -836,7 +837,31 @@ using size_sequence = constant_sequence<std::size_t, Values...>;
...
+                    : (cur * 2 <= end ? Flag::eDouble
+                                      : Flag::eRecurse);
+}
+
+template<std::size_t End,
```
Comment (juchem) [inline fatal/type/sequence.h]: re-format

```
template<
  std::size_t End,
  typename State = size_sequence<0>,
  build_strategy Status = flag(State::size, End)
>
```

File: fatal/type/sequence.h [hunk 4]
```diff
@@ -836,7 +837,31 @@ using size_sequence = constant_sequence<std::size_t, Values...>;
...
+
+template<typename, typename>
+struct concat;
+
+template<typename T, T Offset, typename>
+struct coerce;
+
+template<typename T>
+constexpr std::size_t distance(T begin, T end) {
+    return begin <= end ? static_cast<std::size_t>(end - begin) :
```
Comment (juchem) [inline fatal/type/sequence.h]: re-format. also, indentation is two spaces

```
  return begin <= end
    ? static_cast<std::size_t>(end - begin)
    : throw "The start of the constant_range must be less than or equal to the end";
```

**Comment by yfeldblum** [comment]
+1

**Comment by ericniebler** [comment]
I botched the squash. I'll close this and make a clean pull request.

**Pull request closed without merging**